### PR TITLE
Added the offical test for recipe

### PR DESCRIPTION
**Pull request by sdy623**
1. I added the vaporsynth's offical unittest in the meta.yaml.
2. I also implemented the build-locally.py for windows
3. Re-rendered with conda-build 3.27.0, conda-smithy 3.34.1, and conda-forge-pinning 2024.04.13.08.49.00

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by sdy623** [comment]
@sshockwave

File: recipe/meta.yaml
Comment (sshockwave) [inline recipe/meta.yaml:57]: Didn't conda automatically import this? If not, how did
```
"$PYTHON" -c "from vapoursynth import core; print(core.version())"
```
pass the test?

**Review by sshockwave — CHANGES_REQUESTED** [review]
Also, I can't find `build-locally.py` anywhere. What is it?

File: recipe/meta.yaml
Comment (sdy623) [reply recipe/meta.yaml:32]: I think the git is needed for pulling the codes from git.

File: recipe/meta.yaml
Comment (sshockwave) [inline recipe/meta.yaml:69]: Why do we need to run the unit tests in conda builds? I don't see it in other recipes as the build resource is so precious that [they are not even willing to use git](https://conda-forge.org/docs/maintainer/adding_pkgs/#build-from-tarballs-not-repos).

File: recipe/meta.yaml
Comment (sdy623) [reply recipe/meta.yaml:57]: Actually, the imports in `test` part in meta.yaml, the conda-build package will run the python -c "import vapoursynth" automatically.

File: recipe/meta.yaml
Comment (sdy623) [reply recipe/meta.yaml:69]: I think it is important to run the unit test, It can prevent producing the brocken packages.

**Comment by sdy623** [comment]
> Also, I can't find `build-locally.py` anywhere. What is it?

I forgot to add it in git

File: recipe/bld.bat
Comment (sdy623) [reply recipe/bld.bat:6]: Yes

File: recipe/meta.yaml
Comment (sshockwave) [reply recipe/meta.yaml:68]: Of course, running the unit test is safer in general. However, the unit tests are focused on the program logic, not packaging. Conda-forge is only a distribution project whose computation resource is not meant for helping the software maintainers test their software. Not even for platform-dependent packages like [libhwy](https://github.com/conda-forge/libhwy-feedstock).

If you insist on adding tests, we need more concrete reasons rather than general guidelines. This is beyond my scope of authority and I will have to bring someone from the conda-forge org into the discussion before merging this PR.

**Review by sdy623 — COMMENTED** [review]
> Of course, running the unit test is safer in general. However, the unit tests are focused on the program logic, not packaging. Conda-forge is only a distribution project whose computation resource is not meant for helping the software maintainers test their software. Not even for platform-dependent packages like [libhwy](https://github.com/conda-forge/libhwy-feedstock).

If you insist on adding tests, we need more concrete reasons rather than general guidelines. This is beyond my scope of authority and I will have to bring someone from the conda-forge org into the discussion before merging this PR. 

I get it, when the new version of vs released, the test only run on the local environments and when the locally test passed, the unittest in the build environment will not needed.

File: recipe/meta.yaml
Comment (sshockwave) [reply recipe/meta.yaml:56]: I see. I used `core.version()` to test the package since Vapoursynth automatically loads plugins under `core` and I was concerned whether the plugins could be correctly found. This is slightly different from simply `import vapoursynth`. What do you think?

**Comment by sshockwave** [comment]
`build-locally.py` is automatically generated by ci and should not be edited by hand.

@conda-forge-admin, please rerender

**Comment by sshockwave** [comment]
> I get it, when the new version of vs released, the test only run on the local environments and when the locally test passed, the unittest in the build environment will not needed.

Yes. You are free to test locally, but the tests should not be going into the recipe.

**Comment by sshockwave** [comment]
I'm going to close this PR if there are no further changes. Feel free to reopen if anything comes up. Thank you for your contribution!

**Pull request closed without merging**
